### ISSUE TITLE
use org.springframework.http.HttpStatus instead of 'int response'

### ISSUE THREAD

**Issue by m-schroeer** [togglz/togglz#492]
This is a question first, as it breaks small piece of current code but adds no change in functionality.

Currently the annotation `FeaturesAreActive` is using plain integer as response status (see below).

https://github.com/togglz/togglz/blob/710e75c0322bf28edd03182d5448ea3353702f1d/spring-web/src/main/java/org/togglz/spring/web/FeaturesAreActive.java#L43

I would suggest to use `org.springframework.http.HttpStatus`instead as it is done inside spring as well (see below)
[spring-web/src/main/java/org/springframework/web/bind/annotation/ResponseStatus.java](https://github.com/spring-projects/spring-framework/blob/dd8ea89bea9809e5733eff954a13ab12516f731f/spring-web/src/main/java/org/springframework/web/bind/annotation/ResponseStatus.java#L77):
``` Java
@AliasFor("value")
HttpStatus code() default HttpStatus.INTERNAL_SERVER_ERROR;
```

More over it is kind of used in Togglz test as well:
https://github.com/togglz/togglz/blob/710e75c0322bf28edd03182d5448ea3353702f1d/spring-web/src/test/java/org/togglz/spring/web/FeatureInterceptorTest.java#L96-L99

In my opinion this would provide **consistency** with the spring-framework.

I would provide a PR, if this issue gets positive feedback.
Recommended change in `FeaturesAreActive`:
``` Java
HttpStatus responseStatus() default HttpStatus.NOT_FOUND; // default 404 as before
```

**Comment by bennetelli** [comment]
As it's only used in Spring Web it would be a good idea to align it to be consistent with the spring framework, but:
We use the FeaturesAreActive.responseStatus() in FeatureInterceptor and put it in the HttpServletResponse (response.sendError(featuresAreActiveAnnotation.responseStatus())) which expects an integer.

Not sure if it would work out of the box, but if you want to test it, feel free to do it :+1: 

As long as it keeps the code as simple as it is or even improves the readability, I'm fine with it

**Comment by m-schroeer** [comment]
I've made a PR #496.

As the intended change is really small I've invested slightly more code to mark the previous variant as `@Deprecated`.
There is a method `FeatureInterceptor.getErrorStatus(annotation)` to support side-by-side usage of now deprecated `FeaturesAreActive#responseStatus` and `FeaturesAreActive#errorResponseStatus` (e11d5abe2301c83a2e2d0916fac3ebaeda6f4bd6).

Tests were adapted as well.

--------
FYI:
I noticed (locally and on [travis (builds/754281231)](https://travis-ci.org/github/togglz/togglz/builds/754281231)) that maven tests of [togglz-spring-web](https://github.com/togglz/togglz/tree/master/spring-web) (`mvn test`) are not executed (see image below).

<img width="655" alt="image" src="https://user-images.githubusercontent.com/16744580/104471646-6532b780-55bb-11eb-839a-cbb5cd3f47da.png">.

I have opened an issue for this: #497. By now I had no time to investigate this by myself. Maybe someone has an idea about it.
